hibernate-core/src/main/java/org/hibernate/persister/entity/SingleTableEntityPersister.java

```diff
@@ -759,7 +759,8 @@ public void visitConstraintOrderedTables(ConstraintOrderedTableConsumer consumer
 					tableName,
 					() -> columnConsumer -> columnConsumer.accept(
 							tableName,
-							constraintOrderedKeyColumnNames[tablePosition]
+							constraintOrderedKeyColumnNames[tablePosition],
+							getIdentifierMapping().getJdbcMappings()
```

Comment:
It's probably worth extracing these from the hot loop? I see `getIdentifierMapping()` is typically trivial, but `getJdbcMappings` is sometimes not.

Reply:
Also it's a quadratic loop :/ And I see several `getJdbcMappings` implementations creating defensive list copies.

It seems to call for a change in `getJdbcMappings()` to expose an immutable list by contract, so to avoid all such copies - regardless of this loop.